lib/src/main/scala/spinal/lib/com/jtag/sim/JtagDriver.scala

```diff
@@ -0,0 +1,86 @@
+package spinal.lib.com.jtag.sim
+
+import spinal.core.sim._
+import spinal.lib.com.jtag.Jtag
+
+/*
+    * Jtag Driver tool
+    * Author: Alexis Marquet
+*/
+case class JtagDriver(jtag: Jtag, clockPeriod: Long) {
+  object JtagStates extends Enumeration {
+    val RESET, IDLE,
+    IR_SELECT, IR_CAPTURE, IR_SHIFT, IR_EXIT1, IR_PAUSE, IR_EXIT2, IR_UPDATE,
+    DR_SELECT, DR_CAPTURE, DR_SHIFT, DR_EXIT1, DR_PAUSE, DR_EXIT2, DR_UPDATE, UNKNOWN = Value
+  }
+  var tapState = JtagStates.UNKNOWN
+  def tapStep(tms: Boolean): Unit = {
+    val oldState = tapState
```

Comment:
It is in the current use, but as JtagDriver might be used for other things, it might be interesting to keep track of the FSM?

Reply:
I agree, in that case it would make sense to return it (and also possibly the new state?)
If it's just there that we can do something with it in the future we can add it once we need it.

Reply:
You might have misunderstood ;-)
I just meant to `oldState` val, not the whole FSM...

Reply:
the remark about adding once we need it still made sense! One day we could try to have a jtag path FSM solver, but it's not needed for now, so let's cross that bridge when we get to it.